luxonis_train/core/core.py

```diff
@@ -181,6 +197,7 @@ def __init__(
                 ),
                 pin_memory=self.cfg.trainer.pin_memory,
                 sampler=sampler if view == "train" else None,
+                generator=generator if view in ["val", "test"] else None,
```

Comment:
We want to be very careful here. Generator should only be not None if `self.cfg.trainer.n_validation_batches` is not None and view is `val` or `test`. Current code I believe sets the generator always which would mean that val and test set would be always deterministic (since we manually set the random seed).